ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml

```diff
@@ -17,94 +20,190 @@ import "../stores"
 StatusModal {
     id: popup
 
-    property alias selectedAccount: accountSelector.selectedAccount
+    property var selectedAccount
+    property string networkPrefix: ""
+    property string completeAddressWithNetworkPrefix
+
+    onSelectedAccountChanged: {
+        if (selectedAccount.address) {
+            txtWalletAddress.text = selectedAccount.address
+        }
+    }
+
+    onCompleteAddressWithNetworkPrefixChanged: {
+        qrCodeImage.source = RootStore.getQrCode(completeAddressWithNetworkPrefix)
+    }
 
     //% "Receive"
     header.title: qsTrId("receive")
     contentHeight: layout.implicitHeight
     width: 556
 
+    showHeader: false
+    showAdvancedHeader: true
+
+    hasFloatingButtons: true
+    advancedHeaderComponent: StatusFloatingButtonsSelector {
+        id: floatingHeader
+        itemsModel: RootStore.accounts
+        delegate: StatusClickableTag {
+            maxTextWidth: 110
+            title: name
+            icon.emoji: !!emoji ? emoji: ""
+            icon.emojiSize: Emoji.size.verySmall
+            icon.isLetterIdenticon: !!model.emoji ? true : false
```

Comment:
@Khushboo-dev-cpp you can just have this `icon.isLetterIdenticon: !!model.emoji`